tests/utils.go

```diff
@@ -412,7 +411,24 @@ func (w *ObjectEventWatcher) Watch(ctx context.Context, processFunc ProcessFunc,
 					break
 				}
 			} else {
-				Fail(fmt.Sprintf("unexpected error event: %v", apierrors.FromObject(watchEvent.Object)))
+				switch watchEvent.Object.(type) {
+				case *metav1.Status:
+					status := watchEvent.Object.(*metav1.Status)
+					// api server sometimes closes connections to Watch() client command, so
+					// we attempt re-start the watch
+					//FIXME
+					if status.Message == ApiServerCloseConnectionError {
+						cli.CoreV1().Events(k8sv1.NamespaceAll).
+							Watch(context.Background(), metav1.ListOptions{
```

Comment:
How is this new watch connected to the event watcher outside?I may just overlook the connection.

Reply:
This is a great question because indeed I think that you're right and I forgot to assign it to the `eventWatcher`. The question is why it actually works... I'm trying to figure out a reason why this happens. But definitely it has to be assigned, otherwise the watcher remains dandling.